src/System.Management.Automation/engine/lang/parserutils.cs

```diff
@@ -850,23 +851,42 @@ internal static object ReplaceOperator(ExecutionContext context, IScriptExtent e
             {
                 string lvalString = lval?.ToString() ?? String.Empty;
 
-                // Find a single match in the string.
-                return rr.Replace(lvalString, replacement);
+                return ReplaceOperatorImpl(context, errorPosition, lvalString, rr, substitute);
             }
             else
             {
                 List<object> resultList = new List<object>();
                 while (ParserOps.MoveNext(context, errorPosition, list))
                 {
                     string lvalString = PSObject.ToStringParser(context, ParserOps.Current(errorPosition, list));
-
-                    resultList.Add(rr.Replace(lvalString, replacement));
+                    resultList.Add(ReplaceOperatorImpl(context, errorPosition, lvalString, rr, substitute));
                 }
 
                 return resultList.ToArray();
             }
         }
 
+        /// <summary>
+        /// ReplaceOperator implementation...
```

Comment:
Really we don't need public comment for private method but if we use it please put one final dot in the comment and comments below. Also it is useful to mention that the method is needed to detect a matchEvaluator or use string regex (please reword :) ).